### DISCUSSION TITLE
Add LFO

### DISCUSSION THREAD

**Discussion by GillesLACAUD** [marcel-licence/esp32_basic_synth#7] [Ideas]
You can add some LFO with the help (or not) of the timer (3 Available on the ESP32 i think). You can use the predefine waveform for them but you must include an inverted saw and a S&H waveform

**Comment by ErichHeinemann** [comment]
Hi Gilles, 
I am testing with a LFO to modulate the pitch. I think I could make it available in the next days. Check my fork. I am a bit too stupid to combine it with the code from Marcel.  I am still learning how to deal with Github. I am thinking of a Modulation-Matrix. Modulation of Pitch works already.

**Reply by GillesLACAUD** [reply]
Hi Erich thank you for your message. Me too i am stupit with github :-). I have 2 Lfo in my project with different waveshapes and destinations...I will check if a understand you code and give some informations about mine.
Best

**Reply by marcel-licence** [reply]
I merged in the modulation support from the most recent project today. It uses the sine as "lfo". 

Regarding GitHub, I am also in on the learning curve. It supports very nice and handy features. You can generates branches etc. and also create pull requests. I think in that case I could merge changes back into the master repo

**Comment by ErichHeinemann** [comment]
Hi marcel,
I have used the SINE for LFO too. :)
together with 2 options as additional multipliers -> Modulation speed and Modulation depth ...

I will check Your code and compare it :)

Best Regards
Erich

**Comment by GillesLACAUD** [comment]
Hi Marcel, Hi Erich 


For me i have these differents waveforms for my two LFO 
(ISA is an inverted saw MUT is for mute) 



char Wave_LfoName[WAVE_LFO_COUNT][5] = 
{"SIN","TRI","SAW","ISA","SQU","S&H","NOI","MUT"}; 


And these different destinations 


Amplitue,Cutoff, pitch,Noise volume,Pan,Wave shapping1, Speed LFOx ,Amplitude LFOx 



char Dest_Name[DEST_TYPE_COUNT][MAX_LABEL] = 
{"AMP","CUT","PIT","NOI","PAN","WS1","LxS","LxA"}; 




I will try to share my code.... 


I use Visual code + platform Io with the arduine environnement. 
It is a very good solution to switch betwenn the different files and you have the common utilities for a good text editor (The Arduino is a little bit poor). 
I Have also buy an express JTAG debugger dongle (not testet yet). 


Thank you again Marcel for all the code you share 


Best/ Musically Gilles 


PS My last product with an ESP32/TTGO 


TTGO DRUM MACHINE 













----- Mail original -----

De: "Erich" ***@***.***> 
À: "marcel-licence/esp32_basic_synth" ***@***.***> 
Cc: "Gilles Lacaud" ***@***.***>, "Author" ***@***.***> 
Envoyé: Dimanche 25 Avril 2021 21:40:52 
Objet: Re: [marcel-licence/esp32_basic_synth] Add LFO (#7) 


Hi marcel, 
I have used the SINE for LFO too. :) 
together with 2 options as additional multipliers -> Modulation speed and Modulation depth ... 

I will check Your code and compare it :) 

Best Regards 
Erich

**Reply by marcel-licence** [reply]
You are welcome. I do not use arduino for coding. 
There are some possibilities to share code for sure =)
